microk8s-resources/default-hooks/remove.d/10-cni-link

```diff
@@ -11,8 +11,8 @@ then
       "${SNAP}/sbin/ip" link delete "${link}" || true
     fi
   done
-  for calink in $("${SNAP}/sbin/ip" -j --pretty link show |\
-                 awk '/cali/ {print $2}' | tr -d '",')
+  for calink in $("${SNAP}/sbin/ip" -j link show |\
+                  "${SNAP}/usr/bin/jq" -r '.[].ifname | select(test("cali[a-f0-9]"))')
```

Comment:
One final fix for the regex.
```suggestion
                  "${SNAP}/usr/bin/jq" -r '.[].ifname | select(test("^cali[a-f0-9]*$"))')
```

Reply:
Nice! I've changed it, note I'm using the regex you suggested above, so that `vxlan.calico` is also picked up:

```shell
merkata@mastodon:~/dev/charm-k8s-postgresql$ ip -j link show| /snap/microk8s/current/usr/bin/jq -r '.[].ifname | select(test("^cali[a-f0-9]*$"))'
cali95ad96c92c3
merkata@mastodon:~/dev/charm-k8s-postgresql$ ip -j link show| /snap/microk8s/current/usr/bin/jq -r '.[].ifname | select(test("^vxlan.calico|cali[a-f0-9]*$"))'
vxlan.calico
cali95ad96c92c3
```